compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/FontRasterizationSettings.skiko.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.text
+
+import androidx.compose.ui.text.platform.Platform
+import androidx.compose.ui.text.platform.currentPlatform
+
+/**
+ * Whether edge pixels draw opaque or with partial transparency.
+ */
+@ExperimentalTextApi
+enum class FontSmoothing {
+    /**
+     * no transparent pixels on glyph edges
+     */
+    /**
+     * no transparent pixels on glyph edges
+     */
+    None,
+
+    /**
+     * may have transparent pixels on glyph edges
+     */
+    AntiAlias,
+
+    /**
+     * glyph positioned in pixel using transparency
+     */
+    SubpixelAntiAlias;
+}
+
+/**
+ * Level of glyph outline adjustment
+ */
+@ExperimentalTextApi
+enum class FontHinting {
+    /**
+     * glyph outlines unchanged
+     */
+    None,
+
+    /**
+     * minimal modification to improve constrast
+     */
+    Slight,
+
+    /**
+     * glyph outlines modified to improve constrast
+     */
+    Normal,
+
+    /**
+     * modifies glyph outlines for maximum constrast
+     */
+    Full;
+}
+
+@ExperimentalTextApi
+class FontRasterizationSettings(
+    val smoothing: FontSmoothing,
+    val hinting: FontHinting,
+    val subpixelPositioning: Boolean,
+    val autoHintingForced: Boolean
+) {
+    companion object {
+        val defaultForCurrentPlatform by lazy {
```

Comment:
Not sure about naming here. Probably something like
```suggestion
        val PlatformDefault by lazy {
```

will align better with existing APIs. @igordmn wdyt?

Reply:
`PlatformDefault` looks good to me.